### PR TITLE
[test-integration] Tests stabilizations/fixes

### PR DESCRIPTION
**Pull request by Crumby**
* Fix Credentials class to accept also roles
* Wait for Grafana operator to be running before creating Grafana CRs
* Install latest Grafana CSV rather then specific version
* Fix required role for the grafana-serviceaccount
* Fix cluster pods retrieval
* Fix `extraJvmOpts` field
* Fix label selector for datagrid-service affinity
* Upgrade Infinsipan version to 12.1.3
* XTF framework is now available in maven central

### CODE REVIEW

**Comment by ryanemerson** [comment]
> * Upgrade Infinsipan version to 12.1.3

This should be 12.1.4

**Comment by dmvolod** [comment]
> * Install latest Grafana CSV rather then specific version

We need to remove `spec.name` for GrafanaDashboard CR before. With the latest Grafana operator could be a unpredictable results 
https://github.com/integr8ly/grafana-operator/commit/46555b15f2f598d24c3604874ff68a626b54801a

File: test-integration/operator-tests/src/test/java/org/infinispan/operator/installation/MonitoringStackIT.java [hunk 1]
```diff
@@ -69,15 +71,15 @@ static void prepare() throws IOException {
         grafanaShift.recreateProject();
         grafanaShift.operatorHub().operatorGroups().create(operatorGroup);
         grafanaShift.operatorHub().subscriptions().create(grafanaSubscription);
+        grafanaWaiter.areExactlyNPodsRunning(1, "name", "grafana-operator").waitFor();
```
Comment (dmvolod) [inline test-integration/operator-tests/src/test/java/org/infinispan/operator/installation/MonitoringStackIT.java:74]: @Crumby have you validated with latest (3.10.1) Grafana operator version?

Comment (Crumby) [reply test-integration/operator-tests/src/test/java/org/infinispan/operator/installation/MonitoringStackIT.java:74]: Yes. Works as expected. But the tests themself verify only that Dashboard CR gets created. If you'd like to verify that dashboard works and looks fine you need to go and look

**Review by dmvolod — APPROVED** [review]

**Comment by Crumby** [comment]
> This should be 12.1.4

**Comment by ryanemerson** [comment]
Thanks @Crumby

**Pull request merged**